### PR TITLE
Fix various asset tracker and Thingy:91 build warnings

### PR DESCRIPTION
**Pull request by jtguggedal**
A collection of patches that fixes various compiler and Kconfig warnings.

### CODE REVIEW

File: boards/arm/nrf9160_pca20035/nrf9160_pca20035_defconfig [hunk 1]
```diff
@@ -23,7 +23,6 @@ CONFIG_UART_CONSOLE=y
 
 # Enable I2C
 CONFIG_I2C=y
-CONFIG_I2C_2=y
```
Comment (SebastianBoe) [inline boards/arm/nrf9160_pca20035/nrf9160_pca20035_defconfig]: @anangl : Does this look good to you?

Comment (jtguggedal) [reply boards/arm/nrf9160_pca20035/nrf9160_pca20035_defconfig]: My limited understanding is that this should be okay due to this one enabling `I2C_2` if `I2C` is enabled (or not disabled by MCUboot):
https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/master/boards/arm/nrf9160_pca20035/Kconfig.defconfig#L21

**Comment by jtguggedal** [comment]
@rlubos Candidate for v1.1.0?

File: boards/arm/nrf9160_pca20035/nrf9160_pca20035_defconfig (hunk 1 above)
Comment (anangl) [reply boards/arm/nrf9160_pca20035/nrf9160_pca20035_defconfig]: > @anangl : Does this look good to you?

With the above explanation, it makes sense. Maybe then some comment in this file explaining why `CONFIG_I2C_2=y` is no longer there, could be helpful for the future?

Comment (jtguggedal) [reply boards/arm/nrf9160_pca20035/nrf9160_pca20035_defconfig]: > > @anangl : Does this look good to you?
> 
> With the above explanation, it makes sense. Maybe then some comment in this file explaining why `CONFIG_I2C_2=y` is no longer there, could be helpful for the future?

Yep, that's reasonable, I've added a comment about it.

**Pull request merged**